code/Game.cs

```diff
@@ -24,6 +25,15 @@ public Game()
             }
         }
 
+        public void ChangeRound(BaseRound round)
+        {
+            Assert.NotNull(round);
+
+            Round?.Finish();
```

Comment:
Unless my understanding is completely wrong, but, I'm pretty sure that at any point networked variables can show up as null. This includes fetching all of the players within a game, attempting to fetch out `Game.Instance`, and checking `Round` for example.

The only way we could confirm `Round` not being null would be to setup some `getter` that would automatically assign it to `Waiting` if the fetch for `Round` was null.

Am I wrong with my understanding?

Reply:
Hmmm.... if it have been a round that failed networking in that situation when the round is switching, the entire system would break. That would be a very unreliable system. And afaik the change is triggered if the Round changes, so the Round has to be something before, otherwise a change can't be detected. So theoretically, Round can't be null imo. Otherwise, things would break... But I could be wrong, e.g. if would be weird if the Round is InProgress, a player connects and uses Round.Waiting before receiving a state update...

Reply:
It's not too big of a deal for now, I'll avoid trying to do these null checks for now, we can see if issues arise if not.

Reply:
Nevermind, in this case, a mass number of exceptions if `Round` isn't possible null checked.